File: front-end-gc-main/src/app/entities/department.ts
import { UserFisca } from "./user-fisca";

export class Department {
  idDepartement: number;
  nomDepartement: string;
  descriptionDepartement: string;
  responsableDepartement: UserFisca;
  membresDepartement: UserFisca[];
  
  constructor(
    idDepartement: number,
    nomDepartement: string,
    descriptionDepartement: string,
    responsableDepartement: UserFisca,
    membresDepartement: UserFisca[]
  ) {
    this.idDepartement = idDepartement;
    this.nomDepartement = nomDepartement;
    this.descriptionDepartement = descriptionDepartement;
    this.responsableDepartement = responsableDepartement;
    this.membresDepartement = membresDepartement;
  }
}

File: back-end-gc-main/src/main/java/com/gi/gestioncompetence/entity/Department.java
package com.gi.gestioncompetence.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Department {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long idDepartement;

  private String nomDepartement;
  private String descriptionDepartement;

  @OneToOne
  private UserFisca responsableDepartement;

  @JsonManagedReference
  @OneToMany(cascade=CascadeType.MERGE,fetch = FetchType.LAZY,mappedBy = "departement")
  private List<UserFisca> membresDepartement;
}
